jobs/integration/test_opa.py

```diff
@@ -0,0 +1,419 @@
+                log("Validate that the policy is no longer enforced")
+                await self.validate_create_ns()
+                self._validate_audit_actions(unit)
+            finally:
+                log("Deleting the audit charm")
+                await self.tools.run(
+                    "juju",
+                    "remove-application",
+                    "-m",
+                    self.tools.k8s_connection,
+                    "--force",
+                    "--destroy-storage",
+                    "gatekeeper-audit",
+                )
+                while audit.units:
+                    await asyncio.sleep(5)
+
+            interval = await self.tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                self.tools.k8s_connection,
+                "update-status-hook-interval",
+            )
+            interval = interval[0].strip()
+            try:
+                log("Decrease update status hook interval")
+                await self.tools.run(
+                    "juju",
+                    "model-config",
+                    "-m",
+                    self.tools.k8s_connection,
+                    "update-status-hook-interval=5s",
+                )
+                log("Waiting for status to change to blocked")
+                while (
+                    status := webhook.units[0].workload_status
+                    if webhook.units
+                    else None
+                ) != "blocked":
+                    assert status != "error"
+                    await asyncio.sleep(5)
+                log("Reconcile resources")
+                await unit.run_action("reconcile-resources")
+                while (
+                    status := webhook.units[0].workload_status
+                    if webhook.units
+                    else None
+                ) != "active":
+                    assert status != "error"
+                    await asyncio.sleep(5)
+            finally:
+                await self.tools.run(
+                    "juju",
+                    "model-config",
+                    "-m",
+                    self.tools.k8s_connection,
+                    f"update-status-hook-interval={interval}",
+                )
+
+            # Check that the webhook works
+            await self.validate_create_ns()
+            log("Creating policy and constraint crds")
+            await self.deploy_example_policy()
+            log("Test that the policy is enforced")
+            await self.validate_create_ns_fail()
+            await self.validate_create_ns_with_label()
+            await self.destroy_example_policy()
+        finally:
+            log("Deleting the gatekeeper charm")
+            await self.tools.run(
+                "juju",
+                "remove-application",
+                "-m",
+                self.tools.k8s_connection,
+                "gatekeeper-controller-manager",
+            )
```

Comment:
I delete it because the fixture deletes the applications after the tests have run, but I want the `gatekeeper-controller-manager` gone for the next test to test the audit charm on its own.

Reply:
The `k8s_model` fixture is used in other tests as well, to change the fixture's scope we would need to copy it, I think it's easier to simply clean up the model.